### PR TITLE
TST: Added regression test

### LINKED ISSUE #41382: BUG: DatetimeIndex gets converted to Index when operating on transpose of empty Dataframe
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

```python

#------ pandas 1.2.* ------

import pandas as pd
df = pd.DataFrame(index = pd.DatetimeIndex([]))

print(df.T.sum().index) # output: Index([], dtype='object')
print(df.sum(axis = 1).index) # output: DatetimeIndex([], dtype='datetime64[ns]', freq=None)

```

#### Problem description
In pandas versions >= 1.2, if we take the sum of the transpose of an empty DataFrame with DatetimeIndex, it results in an output with normal Index instead of DatetimeIndex.

But if we sum the DataFrame with axis=1, which is an equivalent operation, the DatetimeIndex is preserved.

#### Expected Output

In pandas versions <= 1.1, the output correctly has a DatetimeIndex in both cases.

```python

#------ pandas 1.1.* ------

import pandas as pd
df = pd.DataFrame(index = pd.DatetimeIndex([]))

print(df.T.sum().index) # output: DatetimeIndex([], dtype='datetime64[ns]', freq=None)
print(df.sum(axis = 1).index) # output: DatetimeIndex([], dtype='datetime64[ns]', freq=None)
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.2.4
numpy            : 1.19.5
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 21.1.1
setuptools       : 56.1.0
Cython           : 0.29.16
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.0
IPython          : 7.23.1
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.2
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.3
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] closes #41382
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/tests/indexing/test_datetime.py [hunk 1]
```diff
@@ -10,6 +10,21 @@
 
 
 class TestDatetimeIndex:
+    def test_datetimeindex_transpose_empty_df(self):
+        """
```
Comment (phofl) [inline pandas/tests/indexing/test_datetime.py:14]: Did you post the link on purpose instead of only the reference like GH#xxx ? If yes, ok for me

**Review by phofl — APPROVED** [review]
small comment, lgtm otherwise

File: pandas/tests/indexing/test_datetime.py (hunk 1 above)
Comment (ShaharNaveh) [reply pandas/tests/indexing/test_datetime.py:14]: Yes, on (some) terminal emulators, clicking on the link will auto open them.
Also when seeing the code via a browser, it's easier to copy it, and paste it in the URL bar instead of: copying the number and then searching it in the issues tab.

---
TL;DR

This is for convenience, but not a deal breaker. I can have it as `GH#XXXX` if it's a deal breaker:)

Comment (phofl) [reply pandas/tests/indexing/test_datetime.py:14]: No strong opinion, would prefer consistency, but if you have a strong optinion this is fine with me

**Comment by datapythonista** [comment]
Thanks for the test @ShaharNaveh
